src/devices/laserFromPointCloud/laserFromPointCloud.cpp

```diff
@@ -0,0 +1,516 @@
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#define _USE_MATH_DEFINES
+
+#include "laserFromPointCloud.h"
+
+#include <yarp/os/Time.h>
+#include <yarp/os/Log.h>
+#include <yarp/os/LogStream.h>
+#include <yarp/os/ResourceFinder.h>
+#include <yarp/math/Math.h>
+
+#include <yarp/rosmsg/sensor_msgs/PointCloud2.h>
+#include <yarp/os/Node.h>
+#include <yarp/os/Publisher.h>
+
+#include <cmath>
+#include <cstring>
+#include <cstdlib>
+#include <iostream>
+#include <limits>
+#include <mutex>
+
+using namespace std;
+
+#ifndef DEG2RAD
+#define DEG2RAD M_PI/180.0
+#endif
+
+/*
+
+yarpdev --device Rangefinder2DWrapper --subdevice laserFromPointCloud \
+        --ROS::useROS true --ROS::ROS_nodeName /cer-laserFront \
+        --ROS::ROS_topicName /laserDepth --ROS::frame_id /mobile_base_lidar_F \
+        --SENSOR \
+        --RGBD_SENSOR_CLIENT::localImagePort    /clientRgbPort:i     \
+        --RGBD_SENSOR_CLIENT::localDepthPort    /clientDepthPort:i   \
+        --RGBD_SENSOR_CLIENT::localRpcPort      /clientRpcPort       \
+        --RGBD_SENSOR_CLIENT::remoteImagePort   /SIM_CER_ROBOT/depthCamera/rgbImage:o \
+        --RGBD_SENSOR_CLIENT::remoteDepthPort   /SIM_CER_ROBOT/depthCamera/depthImage:o \
+        --RGBD_SENSOR_CLIENT::remoteRpcPort     /SIM_CER_ROBOT/depthCamera/rpc:i    \
+        --TRANSFORM_CLIENT::local               /laserFromDepth/tfClient    \
+        --TRANSFORM_CLIENT::remote              /transformServer    \
+        --Z_CLIPPING_PLANES::floor_height       0.15 \
+        --Z_CLIPPING_PLANES::ceiling_height     3.0  \
+        --Z_CLIPPING_PLANES::camera_frame_id depth_center \
+        --Z_CLIPPING_PLANES::ground_frame_id ground_link \
+        --publish_ROS_pointcloud \
+        --period 10 \
+        --name /outlaser:o
+*/
+
+yarp::os::Publisher<yarp::rosmsg::sensor_msgs::PointCloud2>* pointCloud_outTopic = nullptr;
+yarp::os::Node                                             * rosNode=nullptr;
+
+void ros_init_pc()
+{
+    //  rosNode = new yarp::os::Node("laserFromPointCloud");
+
+    pointCloud_outTopic=new yarp::os::Publisher<yarp::rosmsg::sensor_msgs::PointCloud2>;
+    if (pointCloud_outTopic->topic("/ros_pc")==false)
+    {
+        yError() << "opening topic";
+    }
+    else
+    {
+        yInfo() << "topic successful";
+    }
+}
+
+void ros_compute_and_send_pc(const yarp::sig::PointCloud<yarp::sig::DataXYZ>& pc, string frame_id)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Function parameter 'frame_id' should be passed by const reference.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=5034547)